forte/data/ontology/top.py

```diff
@@ -914,223 +980,412 @@ def __init__(
+        return self._cx
 
     @property
-    def center(self):
-        return (self._cy, self._cx)
+    def box_center(self) -> Tuple[int, int]:
+        """
+        Get the box center by using the property function cy() and cx().
+        If box center is not set not computable, it raises a ``ValueError``
+
+        Returns:
+            The box center in a ``Tuple`` format.
+        """
+        return (self.cy, self.cx)
 
     @property
-    def corners(self):
+    def corners(self) -> Tuple[Tuple[int, int], ...]:
         """
-        Get corners of box.
+        Compute and return the corners of the box.
+
+        Raises:
+            ValueError: if the box center is not set.
+
+        Returns:
+            The corners of the box in a ``Tuple`` format.
         """
-        return [
-            (self._cy + h_offset, self._cx + w_offset)
-            for h_offset in [-0.5 * self._height, 0.5 * self._height]
-            for w_offset in [-0.5 * self._width, 0.5 * self._width]
-        ]
+        return tuple(
+            (self.cy + h_offset, self.cx + w_offset)
+            for h_offset in [-self._height // 2, self._height // 2]
+            for w_offset in [-self._width // 2, self._width // 2]
+        )
 
     @property
-    def box_min_x(self):
-        return max(self._cx - round(0.5 * self._width), 0)
+    def box_min_x(self) -> int:
+        """
+        Compute the minimum x coordinate of the box.
+
+        Raises:
+            ValueError: if the box center is not set.
+
+        Returns:
+            The minimum x coordinate of the box.
+        """
+        return max(self.cx - round(0.5 * self._width), 0)
 
     @property
-    def box_max_x(self):
-        return min(self._cx + round(0.5 * self._width), self.max_x)
+    def box_max_x(self) -> int:
+        """
+        Compute the maximum x coordinate of the box.
+
+        Raises:
+            ValueError: if the box center is not set.
+
+        Returns:
+            The maximum x coordinate of the box.
+        """
+        return self.cx + self._width // 2
 
     @property
-    def box_min_y(self):
-        return max(self._cy - round(0.5 * self._height), 0)
+    def box_min_y(self) -> int:
+        """
+        Compute the minimum y coordinate of the box.
+
+        Raises:
+            ValueError: if the box center is not set.
+
+        Returns:
+            The minimum y coordinate of the box.
+        """
+        return max(self.cy - round(0.5 * self._height), 0)
 
     @property
-    def box_max_y(self):
-        return min(self._cy + round(0.5 * self._height), self.max_y)
+    def box_max_y(self) -> int:
+        """
+        Compute the maximum y coordinate of the box.
+
+        Raises:
+            ValueError: if the box center is not set.
+
+        Returns:
+            The maximum y coordinate of the box.
+        """
+        return self.cy + round(0.5 * self._height)
 
     @property
-    def area(self):
+    def area(self) -> int:
```

Comment:
Something like `area` should also exist in the upper classes

Reply:
https://github.com/asyml/forte/issues/863